test/TritonGPU/loop-pipeline.mlir

```diff
@@ -20,17 +20,18 @@
 // CHECK: %[[A1BUFFER:.*]] = triton_gpu.insert_slice_async {{.*}}, {{.*}}, %[[CONSTANT_1]]
 // CHECK: %[[B1BUFFER:.*]] = triton_gpu.insert_slice_async {{.*}}, {{.*}}, %[[CONSTANT_1]]
 // CHECK:   triton_gpu.async_wait {num = 2 : i32}
-// CHECK: %[[A0:.*]] = triton_gpu.extract_slice %[[A1BUFFER]], %[[CONSTANT_0]]
-// CHECK: %[[B0:.*]] = triton_gpu.extract_slice %[[B1BUFFER]], %[[CONSTANT_0]]
+// CHECK: %[[A0:.*]] = tensor.extract_slice %[[A1BUFFER]][0, 0, 0]
```

Comment:
Nice reuse to avoid unnecessary reinventing wheels.

And maybe I am missing something, but where is the code change about from using `triton_gpu.extract_slice` to `tensor.extract_slice`?

Reply:
> Nice reuse to avoid unnecessary reinventing wheels.

Yeah I agree. 

> And maybe I am missing something, but where is the code change about from using triton_gpu.extract_slice to tensor.extract_slice?

That would be on me.

I have the following questions:

- Shouldn't `tensor.extract_slice` accept three parameters instead of two? For instance `%1 = tensor.extract_slice %0[0, 0, 0][1, 16, 4][1, 1, 1]`. While the code above is `tensor.extract_slice %[[A1BUFFER]][0, 0, 0]`.
- Considering that we couldn't fully support `stride` and `size` for performance consideration. Is it still necessary to use the tensor Dialect for now? I was considering about lowering `triton_gpu.extract_slice` to `tensor.extract_slice` once we've designed the intermediate dialect between `triton_gpu` and `llvm`.

Reply:
> Considering that we couldn't fully support stride and size for performance consideration. Is it still necessary to use the tensor Dialect for now?

I think eventually, we do want to support all the features of `tensor.extract_slice`, but you are right that doing so efficiently will require a very careful design. I believe that it's better to just switch to `tensor.extract_slice` now, since we'll probably want to switch eventually.

Reply:
Alright, I will create a patch after Da's PR is merged.